base/logging_test.go

```diff
@@ -242,3 +244,140 @@ func TestLogColor(t *testing.T) {
+
+	logger.level = LevelNone
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.False(t, logger.shouldLog(LevelWarn))
+	assert.False(t, logger.shouldLog(LevelError))
+	assert.False(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+
+	logger.level = LevelWarn
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.True(t, logger.shouldLog(LevelWarn))
+	assert.True(t, logger.shouldLog(LevelError))
+	assert.False(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+
+	logger.level = LevelError
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.False(t, logger.shouldLog(LevelWarn))
+	assert.True(t, logger.shouldLog(LevelError))
+	assert.False(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+
+	logger.level = LevelInfo
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.True(t, logger.shouldLog(LevelWarn))
+	assert.True(t, logger.shouldLog(LevelError))
+	assert.True(t, logger.shouldLog(LevelInfo))
+	assert.False(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+
+	logger.level = LevelDebug
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.True(t, logger.shouldLog(LevelWarn))
+	assert.True(t, logger.shouldLog(LevelError))
+	assert.True(t, logger.shouldLog(LevelInfo))
+	assert.True(t, logger.shouldLog(LevelDebug))
+	assert.False(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+
+	logger.level = LevelTrace
+	assert.True(t, logger.shouldLog(LevelNone))
+	assert.True(t, logger.shouldLog(LevelWarn))
+	assert.True(t, logger.shouldLog(LevelError))
+	assert.True(t, logger.shouldLog(LevelInfo))
+	assert.True(t, logger.shouldLog(LevelDebug))
+	assert.True(t, logger.shouldLog(LevelTrace))
+	out = CaptureOutput(t, LogSyncGatewayVersion)
+	assert.Contains(t, out, LongVersionString)
+}
+
+func CaptureOutput(t *testing.T, f func()) string {
+	reader, writer, err := os.Pipe()
+	assert.NoError(t, err)
+	stdout := os.Stdout
+	stderr := os.Stderr
+
+	defer func() {
+		os.Stdout = stdout
+		os.Stderr = stderr
+		log.SetOutput(os.Stderr)
+	}()
+
+	os.Stdout = writer
+	os.Stderr = writer
+	log.SetOutput(writer)
+
+	out := make(chan string)
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+
+	go func() {
+		var buf bytes.Buffer
+		wg.Done()
+		io.Copy(&buf, reader)
+		out <- buf.String()
+	}()
+
+	wg.Wait()
+	f()
+	writer.Close()
+	return <-out
```

Comment:
A channel, a waitgroup, and a buffer seem unnessesary, but given the suggested `consoleOutput` change above, you won't need to do nearly as much work.

It should be as simple as:

```golang
func CaptureOutput(f func()) string {
	buf := bytes.Buffer{}

	consoleOutput = &buf
	f()
	consoleOutput = os.Stderr

	return buf.String()
}
```

Reply:
Yeah, this makes more sense!